soco/core.py

```diff
@@ -1490,6 +1494,70 @@ def get_item_album_art_uri(self, item):
         else:
             return None
 
+    @only_on_master
+    def set_sleep_timer(self, sleep_time_seconds):
+        """Sets the sleep timer.
+
+        Args:
+            sleep_time_seconds (`int/NoneType`): How long to wait before
+                turning off speaker in seconds, None to cancel a sleep timer.
+                Maximum value of 86399
+
+        Raises:
+            SoCoException: Upon errors interacting with Sonos controller
+            ValueError: Argument/Syntax errors
+
+        """
+        # Note: A value of None for sleep_time_seconds is valid, and needs to
+        # be preserved distinctly separate from 0. 0 means go to sleep now,
+        # which will immediately start the sound tappering, and could be a
+        # useful feature, while None means cancel the current timer
+        if sleep_time_seconds == 0:
+            sleep_time = '00:00:00'
+        elif not sleep_time_seconds:
+            sleep_time = ''
+        elif sleep_time_seconds != int(sleep_time_seconds):
+            raise ValueError('invalid sleep_time_seconds, must be integer \
+                value between 0 and 86399 inclusive')
+        else:
+            sleep_time = str(
+                datetime.timedelta(seconds=int(sleep_time_seconds))
+            )
+
+        try:
+            self.avTransport.ConfigureSleepTimer([
+                ('InstanceID', 0),
+                ('NewSleepTimerDuration', sleep_time),
+            ])
+        except SoCoUPnPException as err:
+            if re.match(r'Error 402 received', str(err)):
```

Comment:
Maybe a regex match is a little heavy handed. If you can find the field on the exception that has the error message, you could do a simple `in` test, i.e: `substring in error_message`